spec/models/dialog_spec.rb

```diff
@@ -68,39 +68,22 @@
 
   context "validate label uniqueness" do
     it "with same label" do
-      expect { @dialog = FactoryGirl.create(:dialog, :label => 'dialog') }.to_not raise_error
-      expect { @dialog = FactoryGirl.create(:dialog, :label => 'dialog') }
+      dialog = FactoryGirl.create(:dialog)
+
+      expect { FactoryGirl.create(:dialog, :name => dialog.name) }
         .to raise_error(ActiveRecord::RecordInvalid, /Name is not unique within region/)
     end
 
     it "with different labels" do
-      expect { @dialog = FactoryGirl.create(:dialog, :label => 'dialog')   }.to_not raise_error
-      expect { @dialog = FactoryGirl.create(:dialog, :label => 'dialog 1') }.to_not raise_error
-    end
-  end
+      FactoryGirl.create(:dialog)
 
-  context "#create" do
-    it "validates_presence_of name" do
-      expect do
-        FactoryGirl.create(:dialog, :label => nil)
-      end.to raise_error(ActiveRecord::RecordInvalid, /Label can't be blank/)
-      expect { FactoryGirl.create(:dialog, :label => 'dialog') }.not_to raise_error
-
-      expect do
-        FactoryGirl.create(:dialog_tab, :label => nil)
-      end.to raise_error(ActiveRecord::RecordInvalid, /Label can't be blank/)
-      expect { FactoryGirl.create(:dialog_tab, :label => 'tab') }.not_to raise_error
-
-      expect do
-        FactoryGirl.create(:dialog_group, :label => nil)
-      end.to raise_error(ActiveRecord::RecordInvalid, /Label can't be blank/)
-      expect { FactoryGirl.create(:dialog_group, :label => 'group') }.not_to raise_error
+      expect { FactoryGirl.create(:dialog) }.to_not raise_error
     end
   end
 
   context "#destroy" do
-    before do
-      @dialog = FactoryGirl.create(:dialog, :label => 'dialog')
+    before(:each) do
+      @dialog = FactoryGirl.create(:dialog, :name => 'dialog')
```

Comment:
Again, I don't think we need to override the `:name` here

Reply:
Also, this doesn't need to be shared.  Can the `create`s move inside the tests themselves?  Then you can get rid of the `before` block entirely.